eq/client/detail/channel.ipp

```diff
@@ -41,45 +46,94 @@ enum State
 class Channel
 {
 public:
-    Channel()
-            : state( STATE_STOPPED )
-            , fbo( 0 )
-            , initialSize( Vector2i::ZERO )
+    Channel( eq::Channel& parent_ )
+        : parent( parent_ )
+        , state( STATE_STOPPED )
+        , fbo( 0 )
+        , initialSize( Vector2i::ZERO )
 #ifdef EQUALIZER_USE_DEFLECT
-            , _dcProxy( 0 )
+        , _dcProxy( 0 )
 #endif
+    {
+        lunchbox::RNG rng;
+        color.r() = rng.get< uint8_t >();
+        color.g() = rng.get< uint8_t >();
+        color.b() = rng.get< uint8_t >();
+    }
+
+    ~Channel()
+    {
+        framebufferImage.flush();
+        statistics->clear();
+        LBASSERT( !fbo );
+    }
+
+    void addResultImageListener( ResultImageListener* listener )
+    {
+        LBASSERT( std::find( resultImageListeners.begin(),
+                             resultImageListeners.end(), listener ) ==
+                  resultImageListeners.end( ));
+
+        resultImageListeners.push_back( listener );
+    }
+
+    void removeResultImageListener( ResultImageListener* listener )
+    {
+        ResultImageListeners::iterator i =
+                std::find( resultImageListeners.begin(),
+                           resultImageListeners.end(), listener );
+        if( i != resultImageListeners.end( ))
+            resultImageListeners.erase( i );
+    }
+
+    void frameViewFinish( eq::Channel* channel )
+    {
+        if( channel->getSAttribute( channel->SATTR_DUMP_IMAGE ).empty( ))
+            removeResultImageListener( &frameWriter );
+        else
         {
-            lunchbox::RNG rng;
-            color.r() = rng.get< uint8_t >();
-            color.g() = rng.get< uint8_t >();
-            color.b() = rng.get< uint8_t >();
+            ResultImageListeners::iterator i =
+                    std::find( resultImageListeners.begin(),
+                               resultImageListeners.end(), &frameWriter );
+            if( i == resultImageListeners.end( ))
+                addResultImageListener( &frameWriter );
```

Comment:
I thought about it too, but the way is initially designed is that one could in theory add and remove the writer at runtime by changing the channel's attribute. If this is not desired or a use case for now, I'm happy to move this to configInit/Exit.